executor/builder.go

```diff
@@ -1556,6 +1556,61 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plannercore.PhysicalIndexJoin)
 	e.innerCtx.keyCols = innerKeyCols
 	e.joinResult = e.newFirstChunk()
 	metrics.ExecutorCounter.WithLabelValues("IndexLookUpJoin").Inc()
+
+	// If the inner join keys are the suffix set of the index, then return the IndexLookUpMergeJoin
+	var isIndexScan bool
+	var is *plannercore.PhysicalIndexScan
+	if ir, ok := innerPlan.(*plannercore.PhysicalIndexReader); ok {
+		is, isIndexScan = ir.IndexPlans[0].(*plannercore.PhysicalIndexScan)
+	}
+	if ilr, ok := innerPlan.(*plannercore.PhysicalIndexLookUpReader); ok {
+		is, isIndexScan = ilr.IndexPlans[0].(*plannercore.PhysicalIndexScan)
+	}
+	if isIndexScan {
+		idx := is.Index.Columns
+		if len(v.InnerJoinKeys) <= len(idx) {
+			isInnerKeysSuffix := true
+			for i, innerKey := range v.InnerJoinKeys {
+				if innerKey.ColName != idx[i].Name {
+					isInnerKeysSuffix = false
+					break
+				}
+			}
+			if isInnerKeysSuffix {
+				is.KeepOrder = true
+				// keepOuterOrder means the outerJoinKeys must be the suffix of the prop items of PhysicalIndexJoin
+				keepOuterOrder := v.KeepOuterOrder && len(v.OuterJoinKeys) < len(v.PropItems)
```

Comment:
What about `len(v.OuterJoinKeys) == len(v.PropItems)`? And when will `len(v.OuterJoinKeys) > len(v.PropItems)`? And if it cannot keep order while its' parent requires, will it lead to wrong result?

Reply:
I'm not sure. Just for safety.

Reply:
Never mind, I misunderstand the `keepOuterOrder`, I thought it was to keep the original order. Maybe `enforceOuterOrder` would be better.